src/constants/oracles.ts

```
@@ -0,0 +1,68 @@
+import { ChainId } from '../crypto/networks';
+import { PartialRecord } from '../types/common';
+import { getOneTokenFactoryContract } from '../utils/contracts';
+import { getProvider } from '../crypto/providers';
+import { OracleName, OracleNames } from '../enums/oracleName';
+import { AddressName } from '../enums/addressName';
+import { getAddress } from './addresses';
+import { OneTokenFactory } from '../generated';
+
+type OracleMapping = {
+  [oracleValues in OracleNames]: PartialRecord<ChainId, string>;
+};
+
+// TODO: I think we can totally remove this and merge it into the TOKENS?  So here are some overlap here, really we want
+// tokens to be in tokens.ts and other contracts are fine to be in here...
+export const ORACLES: OracleMapping = {
+  [OracleName.MAINNET_ICHI_ORACLE]: { [ChainId.Mainnet]: '0xD41EA28e17BD06136c416cA942fB997122138139' },
+};
+
+export async function getAllOneTokenOracles(chainId: ChainId): Promise<Map<string, string[]>> {
+  let oneTokenToOracle = new Map<string, string[]>();
+  const ichiVaultFactory = await getOneTokenFactoryInstance(chainId);
+
+  let numForeignTokens = await ichiVaultFactory.foreignTokenCount();
+  for (let i = 0; i < numForeignTokens.toNumber(); i++) {
+    let foreignToken = await ichiVaultFactory.foreignTokenAtIndex(i);
+    oneTokenToOracle.set(foreignToken, await getOneTokenOracles(foreignToken, chainId, ichiVaultFactory));
+  }
+
+  return oneTokenToOracle;
+}
+
+export async function getOneTokenOracles(oneTokenAddress: string, chainId: ChainId, ichiVaultFactoryContract?: OneTokenFactory): Promise<string[]> {
+  let ichiVaultFactory;
+  if (!ichiVaultFactoryContract) {
+    ichiVaultFactory = await getOneTokenFactoryInstance(chainId);
+  } else {
+    ichiVaultFactory = ichiVaultFactoryContract;
+  }
+
+  let oneTokenToOracle: string[] = [];
+
+  let numOracles = await ichiVaultFactory.foreignTokenOracleCount(oneTokenAddress);
```

what if the `ichiVaultFactory` is undefined at this point?  You should have a check before this and throw an error if it's undefined.  Or better yet in the await to get it, wrap that in a try/catch and throw an error